### PR TITLE
Deprecate (min_distance < 1) and (footprint.size < 2) in peak_local_max

### LINKED ISSUE #4749: Deprecate edge cases in peak_local_max
## Description

While investigating for #3990, I found that `peak_local_max` manages two edge cases that can be deprecated:

- `min_distance=0` and `footprint=None`
- `footprint.size=1`

in this cases `peak_local_max` is equivalent to applying a threshold. I think that these settings must be forbidden.

## Version information

```python
3.7.5 (default, Oct 25 2019, 15:51:11) 
[GCC 7.3.0]
Linux-4.4.0-179-generic-x86_64-with-debian-stretch-sid
scikit-image version: 0.18.dev0
numpy version: 1.18.4

```

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Closes #4749

In  these two deprecated cases, `peak_local_max` acts as thresholding:
```python
>>> from skimage.feature import peak_local_max
>>> img = np.random.rand(50, 50)
>>> thresh = 0.2
>>> np.all((img > thresh) == peak_local_max(img, min_distance=0, threshold_abs=thresh,
                                            indices=False, exclude_border=False))
True
>>> np.all((img > thresh) == peak_local_max(img, footprint=np.array([[1]]),
                                            threshold_abs=thresh, indices=False,
                                            exclude_border=False))
True
```

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [x] Unit tests
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/feature/peak.py
Comment (sciunto) [inline skimage/feature/peak.py:168]:
```suggestion
        warn("min_distance lower than 1 is deprecated. "
```

File: skimage/feature/peak.py
Comment (sciunto) [inline skimage/feature/peak.py:175]:
```suggestion
        warn("footprint.size lower than 2 is deprecated. "
```

File: skimage/feature/tests/test_peak.py
Comment (sciunto) [inline skimage/feature/tests/test_peak.py:346]:
```suggestion
        with expected_warnings(["footprint.size lower than 2 is deprecated."]):
```

File: skimage/feature/tests/test_peak.py
Comment (sciunto) [inline skimage/feature/tests/test_peak.py:353]:
```suggestion
        with expected_warnings(["footprint.size lower than 2 is deprecated."]):
```

File: skimage/feature/tests/test_peak.py
Comment (sciunto) [inline skimage/feature/tests/test_peak.py:400]:
```suggestion
        with expected_warnings(["min_distance lower than 1 is deprecated."]):
```

**Review by sciunto — APPROVED** [review]

**Comment by sciunto** [comment]
@scikit-image/core We need a second review over here.

@rfezzani Could you rebase please?

File: skimage/feature/peak.py
Comment (alexdesiqueira) [inline skimage/feature/peak.py:185]: `"In this case (...)`

**Comment by alexdesiqueira** [comment]
Thank you for that @rfezzani! I have a question, though. Wouldn't it be better for the user to still have the option to put these values, receiving a warning instead of a `ValueError`? I would personally think in putting a `Notes` section explaining what happens for `min_distance < 1`, `footprint.size < 2`, and then presenting the warnings you suggested when the user uses these arguments. My two BRL cents, though!

**Comment by rfezzani** [comment]
> Wouldn't it be better for the user to still have the option to put these values, ...

That's in fact another option @alexdesiqueira, I am OK with it.
@scikit-image/core, any opinion on this: deprecation or simple warning + doc?

**Comment by emmanuelle** [comment]
@rfezzani is this idea of forbidding these modes (when the deprecation cycle is complete) that it is inefficient to use `peak_local_max` for something which can be done in a faster and clearer way? 

I think I'd prefer a simple warning here, because if for some reason users want to iterate over the values of `footprint` or `min_distance` it's not very nice to have them add an if/else block to handle the corner cases (since the output of `peak_local_max` is valid).

**Comment by rfezzani** [comment]
@alexdesiqueira and @emmanuelle, I replaced the deprecation by a simple `RuntimeWarning`.

File: skimage/feature/peak.py
Comment (grlee77) [reply skimage/feature/peak.py]: Perhaps not intentionally, but previously an array-like footprint did work. (e.g. `footprint=[[0, 1, 0], [1, 1, 1], [0, 1, 0]]`), but now this would result in an AttributeError. The docstring does indicate that `footprint` should be an ndarray, though, so perhaps it is okay to leave it as is.

If we want to maintain array-like support, use something like:
```Python
if footprint is not None:
    footprint = np.asarray(footprint)
    if footprint.size == 1:
        warn(...
```

**Review by grlee77 — APPROVED** [review]
one comment to consider about array-like footprint, but otherwise this seems good to go

File: skimage/feature/peak.py
Comment (rfezzani) [reply skimage/feature/peak.py]: Good catch, I will keep the `array-like` support :wink:

**Comment by rfezzani** [comment]
Thank you @grlee77! I kept the support for array-like `footprint` as you suggested.

**Review by jni — APPROVED** [review]
In the long run, we should use the warnings to point to the alternative `local_maxima`. But this shouldn't hold up this improvement any longer — that can come in a later PR. Thank you @rfezzani!

**Comment by rfezzani** [comment]
:tada: Thank you @jni

**Pull request merged**
